checker.c

```diff
@@ -1,21 +1,102 @@
+/* Include Headerfiles													*/
+/* ===================													*/
+
 #include <stdio.h>
-#include <assert.h>
+#include <assert.h>	
+#include <stdbool.h>
+#include "checker.h"
+
+
+
+/********************************************************************************
+ * A common function that checks if the paramters are within the boundary conditions.
+ * input: parameter value, Maximum and minimum range to be checked
+ * returns: Check if the parameter is within in the given maximum and minimum range
+ *********************************************************************************/
 
-int batteryIsOk(float temperature, float soc, float chargeRate) {
-  if(temperature < 0 || temperature > 45) {
-    printf("Temperature out of range!\n");
-    return 0;
-  } else if(soc < 20 || soc > 80) {
-    printf("State of Charge out of range!\n");
-    return 0;
-  } else if(chargeRate > 0.8) {
-    printf("Charge Rate out of range!\n");
-    return 0;
-  }
-  return 1;
+bool BMS_AttributeinRangeStatus(float parametervalue, int index)
+
+{	bool InRange=true;
+	if ((parametervalue< (AttributeMinimumthreshold[index]))||(parametervalue>= (AttributeMaximumthreshold[index])))
```

Comment:
This function can be written in a single line:
`return (parametervalue< (AttributeMinimumthreshold[index]))||(parametervalue>= (AttributeMaximumthreshold[index]))`
Do you think that would be more readable?